Update README with regards to GitLab etc

plajjan opened the pull request:
This is the kind of commit I don't personally like. It mixes content changes
with formatting changes (line wrapping, tab -> spaces in lists etc) making it
harder to review. Apologies.

- improved wording in a lot of place
- tried to make text more concise, yet it is longer now as there is more
  information.
- encourage GitLab over GitHub
  - inline with what has been discussed among librarians
- remove mention of DCO compliance checking - there is no bot that forces
  compliance on Git(Lab|Hub) so no reason mentioning it

plajjan committed:
Update README with regards to GitLab etc

This is the kind of commit I don't personally like. It mixes content changes
with formatting changes (line wrapping, tab -> spaces in lists etc) making it
harder to review. Apologies.

- improved wording in a lot of place
- tried to make text more concise, yet it is longer now as there is more
  information.
- encourage GitLab over GitHub
  - inline with what has been discussed among librarians
- remove mention of DCO compliance checking - there is no bot that forces
  compliance on Git(Lab|Hub) so no reason mentioning it

plajjan commented:
@JohanVikman / @jabelk yo, plxz to read!

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
+use the integrated CI system, thus providing a standardized CI configuration
+format, yet setup a customized CI runner (the actual machine running CI jobs).
+This is the primary reason to why GitLab is the preferred location - having a
+specialized CI runner is allows for running NSO, netsims and virtual routers.
```
Redundant "is" here.

...specialized CI runner *is* allows...

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
+This is the primary reason to why GitLab is the preferred location - having a
+specialized CI runner is allows for running NSO, netsims and virtual routers.
+
+NSO-Developer originally started solely on GitHub. GitLab however has become the
```
Maybe skip "solely".

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
+This is the primary reason to why GitLab is the preferred location - having a
+specialized CI runner is allows for running NSO, netsims and virtual routers.
+
+NSO-Developer originally started solely on GitHub. GitLab however has become the
+primary and encouraged place to use for development as the CI functionality
+offered by GitLab is well suited for testing NSO. Specifically, the CI
+functionality of GitLab together with the [NSO in Docker
+project](https://gitlab.com/nso-developer/nso-docker/) allows anyone to very
+easily achieve a development environment for NSO packages or entire NSO systems
+that are both easily tested locally and in CI.
```
We are kind of repeating ourselves here from the earlier paragraph, really pressing on the CI functionality. Maybe just say that we will keep Github because that is where it started and we do not want to force anyone to change?

jabelk replied:
I think the additional explanation is useful, especially since ALL other DevNet content is on Github and not Gitlab. I almost would mention that we have a production grade NSO  CI/CD project already built around Gitlab ecosystem so it makes sense to align to that. Github has its own CI solutions, so I think "well suited" is too polite for people who don't understand the background.

plajjan replied:
I dunno, this is just like an introduction text and a hint to why things are the way they are. I don't feel we need to overdo the convincing part - we can also just talk to the people involved and given deeper explanation :)

plajjan replied:
But yes, it is repetition. I actually wrote the second paragraph first and then went about seeing things from a slightly different perspective (like people aren't interested in history first but rather what it means to them right here and now) and then wrote the first paragraph. I'll rephrase this.

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
 ## Licenses
-All material on the NSO Developer sapce on GitHub is under the [Apache 2.0 license](https://github.com/NSO-developer/NSO-developer/blob/master/LICENSE). 
-The license is used to ensure a balance between open contribution and allowing you to use the software as you like to.
+All material on the NSO-Developer space is under the [Apache 2.0
```
Good change (how did we miss the sapce typo before???).

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
+privileges, and as such you can decide to give others write access for example.
 
-Users of your repository can use Issues to report bugs and suggest new features, and Pull Requests to contribute code.
+Users of your repository can use Issues to report bugs and suggest new features,
```
Bold on "Issues" I think.

JohanVikman replied:
And at the "Merge Requests"

plajjan replied:
Okay, not sure why we capitalize this or why it should be bold.. but I'll roll with that :)

JohanVikman commented on README.md:
```diff
@@ -1,74 +1,130 @@
+When/if you do not have time to keep your project up to date (fix issues, accept
+merge/pull requests etc) - please say so. Write a line in the README, as well as
+an email to nso-developer@cisco.com - we will try to help you find a new
+maintainer of the code, or retire it from the library if it appears abandoned
```
We should abandon some projects... :/

jabelk replied:
Should we add a timeline to say a project is abandoned? Like 18 months of no commit?

plajjan replied:
While this is a change in my MR, I haven't actually changed the text, just the formatting of it, like line-wrapping etc. This is a good conversation, but this MR is the wrong place. Open issue to discuss :)

JohanVikman commented:
@plajjan we should mention that a repository description should be added. So that we don't have a cryptic name without a description.

plajjan committed:
Improve wording and require description

Avoid repetition across paragraphs on why GitLab is better than GitHub.

Added section encouraging users to fill in repository description since that's
quite helpful for anyone browsing through the list of repos.

plajjan commented:
I believe I have addressed all the comments. Please have a look at the updated content!

plajjan commented:
@JohanVikman @jabelk any further comments or good to merge? :)

JohanVikman commented:
Looks good!

plajjan commented:
I'll go ahead and merge this. We can always update later on if there's anything we missed here.
